tests/testTriggers.go

```diff
@@ -3698,6 +3702,96 @@ func TriggerAutoFsTrim(contexts *[]*scheduler.Context, recordChan *chan *EventRe
 
 }
 
+// TriggerVolumeUpdate enables to test volume update
+func TriggerVolumeUpdate(contexts *[]*scheduler.Context, recordChan *chan *EventRecord) {
+	defer ginkgo.GinkgoRecover()
+	event := &EventRecord{
+		Event: Event{
+			ID:   GenerateUUID(),
+			Type: UpdateVolume,
+		},
+		Start:   time.Now().Format(time.RFC1123),
+		Outcome: []error{},
+	}
+	defer func() {
+		event.End = time.Now().Format(time.RFC1123)
+		*recordChan <- event
+	}()
+	context("Validate update of the volumes", func() {
+
+		Step("Update Io priority on volumes ",
+			func() {
+				updateIOPriorityOnVolumes(contexts, event)
+				logrus.Info("Update IO priority call completed")
+			})
+	})
+}
+
+//This method is responsible for updating IO priority on Volumes if volume is labelled with
+//IO Priority. If lebelled with High and volume is updated to HIGH, next iteration it will reset
+//back to low.
+func updateIOPriorityOnVolumes(contexts *[]*scheduler.Context, event *EventRecord) {
+	for _, ctx := range *contexts {
+		var appVolumes []*volume.Volume
+		var err error
+		appVolumes, err = Inst().S.GetVolumes(ctx)
+		UpdateOutcome(event, err)
+		if len(appVolumes) == 0 {
+			UpdateOutcome(event, fmt.Errorf("found no volumes for app "))
+		}
+		for _, v := range appVolumes {
+			isPureVol, err := Inst().V.IsPureVolume(v)
+			if err != nil {
+				UpdateOutcome(event, err)
+			}
+			if isPureVol {
+				logrus.Warningf(
+					"Autofs Trim is not supported for Pure DA volume: [%s]",
+					"Skipping autofs trim status on pure volumes", v.Name,
+				)
+				continue
+			}
+			logrus.Infof("Getting info from volume: %s", v.ID)
+			appVol, err := Inst().V.InspectVolume(v.ID)
+			if err != nil {
+				logrus.Errorf("Error inspecting volume: %v", err)
+			}
+			if requiredPriority, ok := appVol.Spec.VolumeLabels["priority_io"]; ok {
+				if !setIoPriority {
+					if requiredPriority == "low" {
+						requiredPriority = "high"
+					} else if requiredPriority == "high" {
+						requiredPriority = "low"
+					}
+				}
+				logrus.Infof("Expected Priority %v", requiredPriority)
+				logrus.Infof("COS %v", appVol.Spec.GetCos().SimpleString())
+				if !strings.EqualFold(requiredPriority, appVol.Spec.GetCos().SimpleString()) {
+					err = Inst().V.UpdateIOPriority(v.ID, requiredPriority)
+					if err != nil {
+						UpdateOutcome(event, err)
+					}
+					//Verify Volume set with required IOPriority.
+					appVol, err := Inst().V.InspectVolume(v.ID)
+					logrus.Infof("COS after update %v", appVol.Spec.GetCos().SimpleString())
```

Comment:
This log will be more useful if you also add the volume name for which the COS has been updated.

Reply:
We pick one volume at a time and we print volume name at the beginning of the loop iteration, whatever update happens that is all on the same volume.
Logs would look like this 

23:18:06 INFO[2022-07-09 06:18:06] Getting info from volume: pvc-ee6ccdee-3c19-4839-8eba-b1fffa6d4709 
23:18:06 INFO[2022-07-09 06:18:06] Expected Priority Changing to high           
23:18:06 INFO[2022-07-09 06:18:06] Expected Priority high                       
23:18:06 INFO[2022-07-09 06:18:06] COS low                                      
23:18:06 INFO[2022-07-09 06:18:06] Cmmand passed pxctl volume update  --io_priority high  pvc-ee6ccdee-3c19-4839-8eba-b1fffa6d4709  
23:18:07 INFO[2022-07-09 06:18:07] COS after update high                        
23:18:07 INFO[2022-07-09 06:18:07] Update IO priority on [pvc-ee6ccdee-3c19-4839-8eba-b1fffa6d4709] : [high]